client/components/data/query-users/index.jsx

```diff
@@ -37,12 +39,18 @@ class QueryUsers extends Component {
 	}
 
 	request() {
-		this.props.requestUsers( this.props.siteId, this.props.userIds );
+		const { siteId, userIds, currentUserId } = this.props;
+		if ( userIds.length ) {
+			const requestIds = userIds.filter( id => id !== currentUserId );
```

Comment:
Ensure that we're not requesting the current user but just the other authors whose information we're lacking.